storage/innobase/ibuf/ibuf0ibuf.cc

```diff
@@ -2326,9 +2327,8 @@ static void ibuf_delete_recs(const page_id_t page_id)
 loop:
   btr_pcur_t pcur;
   ibuf_mtr_start(&mtr);
-  if (btr_pcur_open(ibuf.index, &tuple, PAGE_CUR_GE, BTR_MODIFY_LEAF,
-                    &pcur, &mtr) != DB_SUCCESS)
-    goto func_exit;
+  btr_pcur_open_on_user_rec(ibuf.index, &tuple, PAGE_CUR_GE,
+                            BTR_MODIFY_LEAF, &pcur, &mtr);
```

Comment:
I had a look of the error handling in 10.6 but it would be a large change to port all part and could impact even non-ibuf part of the code. The interface for opening on user record and placing on next user record (moving to next page) is used in many places (10.5) and extending the usage here (a rarer case) seems safe and inline with 10.5.

Reply:
Added minimal error handling for btr_pcur_open_on_user_rec.